### ISSUE TITLE
Issue: Charging CouchPotato does not work

### ISSUE THREAD

**Issue by sp00kie** [SpaceK33z/movieo-to-plex#168]
1. Connect to Plex.
2. Select server.
3. Enter couchpotato settings.
4. save settings.
5. Go to IMDB and find a movie already in the library.
6. Instead of a "View in Plex" link there is a "Download" button.

It doesn't matter how long you wait. I checked after 12 hrs and refreshed the imdb page and it still had a "Download" button

**Comment by Ephellon** [comment]
Hi. CouchPotato does not cache (**charge**) found media correctly, see #162

> Alright, got the movies to be added again. But, the charging statement does not work (still).
> 
> **Charging** — Web to Plex's "caching" ability; it grabs all movies/shows that the user has already added via their manager(s)

**Comment by sp00kie** [comment]
Sorry, had no idea this was a dupe.  I did search! Also, thanks for the great extension.

**Comment by Ephellon** [comment]
No problem. I'll keep this as a separate ticket, I'll focus on CouchPotato as a whole in #162

**Comment by sp00kie** [comment]
I don't know if this is of any help to you but there is an open source project called ombi that pulls data from plex for basically the same use as your plugin.

**Comment by Ephellon** [comment]
Yes, Ombi support was added to this project already.

**Comment by Ephellon** [comment]
Wait, what version are you using?

**Comment by Ephellon** [comment]
Current version is [ `v4.1.2.4`](https://github.com/SpaceK33z/web-to-plex/releases/tag/v4.1.2.4-1900MST19FEB2020)

**Comment by Ephellon** [comment]
@sp00kie &uarr;

**Comment by sp00kie** [comment]
hi.  I'm using version 3.4. That's what was on the chrome store like 4 days ago.  ok I found version 4.1 and installed it.  It works perfectly for movies.  I'm not sure if its pulling data from couchpotato or plex.  My problem with medusa is that it forces me to select one folder.  The problem is that my tv shows are on 3 different hdds.

**Comment by sp00kie** [comment]
Also, I mentioned ombi because I thought you could use their code for getting plex data because I tested it and it is working.

**Comment by Ephellon** [comment]
Hi @sp00kie,

1. *Pulling*: It's pulling from CouchPotato (**charging**)
2. *Multiple folders*: I have a similar setup; I've already started implementing the `Ctrl` `Click` method for adding multiple folders (in the upcoming version)
3. *Ombi*: I've already added their code, not sure what you mean. As far as getting data from Plex, it currently works just fine; if you're using Ombi, the extension doesn't bother doing anything other than telling Ombi to add the item

**Comment by sp00kie** [comment]
yeah the current version seems to be pulling data from everywhere properly.  Sorry for bothering you with the old version.  I do have a different problem with the current version though.  It changes the layout of websites its working on.  I have attached some examples. It makes my ombi unusable because I can't access settings. The second screenshot is get info on a movie in plex.
<img width="703" alt="Screenshot_3" src="https://user-images.githubusercontent.com/7481596/77849828-26bc3f80-71d7-11ea-9667-7b43335960c1.png">
<img width="538" alt="Screenshot_2" src="https://user-images.githubusercontent.com/7481596/77849830-2754d600-71d7-11ea-853f-938e91c128e5.png">
<img width="271" alt="Screenshot_1" src="https://user-images.githubusercontent.com/7481596/77849832-27ed6c80-71d7-11ea-9f10-e42bc13ee077.png">
<img width="253" alt="Screenshot_17" src="https://user-images.githubusercontent.com/7481596/77849834-27ed6c80-71d7-11ea-84ce-07f87f9cffd6.png">

**Comment by Ephellon** [comment]
Try reinstalling the extension, see #165

**Comment by sp00kie** [comment]
ok so I reinstalled the extension and then restarted chrome.  Whenever I open an imdb page, the extension opens its settings page and switches to it.  If I switch back to imdb and refresh, it will switch back to the extension settings page.  If I have closed it, it will reopen it. 

All pages look fine. Signing into plex and saving settings results in my ombi page being broken like in the screenshot but the imdb page is still fine.  Selecting imdb in default sites and refreshing results in the imdb page being broken including the web to plex button as well. I've tried uninstalling and reinstalling several times with the same results. I also tried doing it with all my other extensions disabled. Exactly the same result.

Also, the firefox and opera extension links on the page below result in it not being found.
https://github.com/SpaceK33z/web-to-plex

I wanted to see if I could get it working properly in another browser.

**Comment by Ephellon** [comment]
<details>
<summary>Sending me the errors (click to expand)</summary>

1. Please enable `Developer Mode` in Web to Plex's settings, and navigate to IMDb or Ombi
2. Press  `Ctrl` `Shift` `J`
3. Select the `Web to Plex` context from the context drop-down (left-hand side)
> ![image](https://user-images.githubusercontent.com/8632548/77867707-9d683580-71f5-11ea-8d6c-579eac5e67c0.png)
4. Open the `Console settings` (right-hand side)
5. Ensure the `Selected context only` box is checked
> ![image](https://user-images.githubusercontent.com/8632548/77867762-def8e080-71f5-11ea-9913-b0798bb7d26d.png)
6. Please send a screenshot of the page and error messages, like so ("The Hunt (2020)" on IMDb) &rarr;
> ![image](https://user-images.githubusercontent.com/8632548/77867913-7e1dd800-71f6-11ea-991a-53ec4adff012.png)
> ![image](https://user-images.githubusercontent.com/8632548/77867905-72caac80-71f6-11ea-814a-3e013b0fdc9c.png)

</details>

----

<details>
<summary>Response to your last post (click to expand)</summary>

1. Settings will nag you until you save them (getting rid of that)
2. I'll investigate with a fresh install
3. Opera works in dev mode only. Firefox was disabled (see #167)

</details>

**Comment by Ephellon** [comment]
@sp00kie &uarr;

Thanks for your help so far. I haven't been able to recreate the errors yet, could you take a look at the post above so I can figure out what's wrong?

**Comment by sp00kie** [comment]
ok that is really weird.  As soon as I turn on developer mode and refresh the IMDB page the format is fine. I'm gonna do it without refreshing the page.

**Comment by Ephellon** [comment]
You won't be able to, the errors only propagate with `Developer mode` enabled

**Comment by Ephellon** [comment]
Thanks for letting me know though... I'll see what's different with `Developer mode`

**Comment by Ephellon** [comment]
It should've just made error messages visible...

**Comment by Ephellon** [comment]
Alright, for now I'll assume remote styles are the problem, that's the only explanation

**Comment by sp00kie** [comment]
here is a screenshot. hope this helps. This is how it looks with the developer mode enabled but not refreshing the page.
<img width="951" alt="Screenshot_5" src="https://user-images.githubusercontent.com/7481596/77869730-3aec5500-7248-11ea-8b68-cf2b36db2d43.png">

**Comment by sp00kie** [comment]
and this is how it looks with the page refreshed after the developer mode has been enabled.
<img width="955" alt="Screenshot_6" src="https://user-images.githubusercontent.com/7481596/77870006-f57c5780-7248-11ea-9e3e-597a1d942265.png">

**Comment by Ephellon** [comment]
Got it! I'm able to reproduce the error now.

**Comment by Ephellon** [comment]
Fixed! Thanks @sp00kie

**Comment by Ephellon** [comment]
Unfortunately, I was in the middle of creating a new version, so I won't be able to update the extension globally.

If you'd like the fix, you can do one of the following:

1. Keep `Developer mode` enabled (highly recommended)
2. [Install this ZIP](https://github.com/SpaceK33z/web-to-plex/files/4400654/src.zip) &mdash; see [installing source code packages](https://github.com/SpaceK33z/web-to-plex/wiki/Installation-Guide#installing-web-to-plex-using-source-code-packages)<sup>1</sup>

1 This package has **not** been thoroughly tested, and contains several features not meant to be available yet. It will break when attempting to use said features.

**Comment by sp00kie** [comment]
Well on my end, Ombi and Couchpotato are fixed but IMDB looks exactly the same. Running version 4.2
<img width="956" alt="Screenshot_9" src="https://user-images.githubusercontent.com/7481596/77894732-5ffbba80-727e-11ea-8214-6c2be15d1a6d.png">

**Comment by Ephellon** [comment]
If you don't mind, change the button shape to `Box`

**Comment by sp00kie** [comment]
When I change it to square, visually IMDB  looks fine even when developer mode is disabled but there is no web-to-plex button that I can see on the page.

**Comment by Ephellon** [comment]
> It's at the top of the page, in the center
> ![image](https://user-images.githubusercontent.com/8632548/77942911-9a168d80-7279-11ea-9b00-ec6c9cf16cff.png)

**Comment by sp00kie** [comment]
omg yeah its right there.  The page looks fine and it works. AWESOME!!!

**Comment by Ephellon** [comment]
No problems noticed. Will close in 4 days

**Comment by Ephellon** [comment]
Not currently in build, but will leave open as reminder
